backend/metrics/calculations.py

```diff
@@ -169,7 +169,7 @@ def extend_and_impute_metrics(
     for column in ['delta.positivity', 'delta.hospitalized', 'delta.active'
         , 'delta.confirmed.ratio_per_million', 'delta.deceased.ratio_per_million',
                    'total.confirmed.ratio_per_million', 'total.deceased.ratio_per_million']:
-        df.loc[:, f'MA.21.{column}'] = _moving_average_grouped(df, grouping_columns, column, rolling_window)
+        df.loc[:, f'MA.21.{column}'] = _moving_average_grouped(df.fillna(0), grouping_columns, column, rolling_window)
```

Comment:
This is a very broad, and hidden fillna(0)

Perhaps just do this on the column that requires this, and perhaps write a comment why 0 is a decent imputation for nan. 

for some of these columns for which the ma.21 is calculated it will skew the curves in a more positive (reduction of covid19) values which might not be desirable since it is not reality. Repeating the previous value might  actually be better in most cases, for an imputed value.

Reply:
We did replaced NA with 0 in non computed columns, but this is missing in the computed column. Probably I can make change that to make it more explicit.

Regarding changing for columns that are in questions- We noticed that for delta positivity MA calculation are not correct which is due to NA values present in positivity column. Same would be issue with other MA calculations as well (haven't come to our notice though).

If delta.tested is NA for one day then 21.MA is actually a 20 day MA instead of 21. Now if tested values are missing for one day that implies they will be added in the next day figures, hence using 0 instead of NA feels correct. 
We do need to fix the MA calculation for these calculated column as otherwise it is actually showing higher figures.